src/amo/components/AddonDetail/index.js

```diff
@@ -240,14 +243,24 @@ export class AddonDetailBase extends React.Component {
   }
 }
 
-export function mapStateToProps(state) {
+export function mapStateToProps(state, ownProps) {
+  const { slug } = ownProps.params;
+  const addon = state.addons[slug];
```

Comment:
When this is saga of course these will have to accept `state.addons[slug]` being empty but I know that's for a future patch 😄